packages/contracts-bedrock/contracts/universal/CrossDomainMessenger.sol

```diff
@@ -370,14 +370,20 @@ abstract contract CrossDomainMessenger is
      * @return Amount of gas required to guarantee message receipt.
      */
     function baseGas(bytes calldata _message, uint32 _minGasLimit) public pure returns (uint32) {
-        return
-            // Dynamic overhead
-            ((_minGasLimit * MIN_GAS_DYNAMIC_OVERHEAD_NUMERATOR) /
-                MIN_GAS_DYNAMIC_OVERHEAD_DENOMINATOR) +
-            // Calldata overhead
-            (uint32(_message.length) * MIN_GAS_CALLDATA_OVERHEAD) +
-            // Constant overhead
+        // We peform the following math on uint64s to avoid overflow errors. Multiplying by
+        // MIN_GAS_DYNAMIC_OVERHEAD_NUMERATOR would otherwise limit the _minGasLimit to
+        // approximately 4.2 MM.
+        // First component is dynamic overhead
```

Comment:
Can we put this in an `unchecked` block since we now have the require at the end?

Reply:
Yeah, good call out